cmd/oras/root/discover.go

```diff
@@ -78,29 +78,34 @@ Example - Discover referrers of the manifest tagged 'v1' in an OCI image layout
 			if err := oerrors.CheckMutuallyExclusiveFlags(cmd.Flags(), "format", "output"); err != nil {
 				return err
 			}
+			opts.RawReference = args[0]
+			if err := option.Parse(cmd, &opts); err != nil {
+				return err
+			}
 			if cmd.Flags().Changed("output") {
-				switch opts.Template {
+				switch opts.Format.Type {
 				case "tree", "json", "table":
 					fmt.Fprintf(cmd.ErrOrStderr(), "[DEPRECATED] --output is deprecated, try `--format %s` instead\n", opts.Template)
```

Comment:
The error message is not updated
```suggestion
					fmt.Fprintf(cmd.ErrOrStderr(), "[DEPRECATED] --output is deprecated, try `--format %s` instead\n", opts.Format.Type)
```